src/test/storage/reference_column_test.cpp

```diff
@@ -0,0 +1,102 @@
+#include <iostream>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "operators/abstract_operator.hpp"
+#include "operators/get_table.hpp"
+#include "operators/print.hpp"
+#include "operators/table_scan.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class ReferenceColumnTest : public ::testing::Test {
+  virtual void SetUp() {
+    _test_table = std::make_shared<opossum::Table>(opossum::Table(3));
+    _test_table->add_column("a", "int");
+    _test_table->add_column("b", "float");
+    _test_table->append({123, 456.7f});
+    _test_table->append({1234, 457.7f});
+    _test_table->append({12345, 458.7f});
+    _test_table->append({54321, 458.7f});
+    _test_table->append({12345, 458.7f});
+
+    _test_table_dict = std::make_shared<opossum::Table>(5);
+    _test_table_dict->add_column("a", "int");
+    _test_table_dict->add_column("b", "int");
+    for (int i = 0; i <= 24; i += 2) _test_table_dict->append({i, 100 + i});
+
+    _test_table_dict->compress_chunk(ChunkID(0));
+    _test_table_dict->compress_chunk(ChunkID(1));
+
+    StorageManager::get().add_table("test_table", _test_table);
+    StorageManager::get().add_table("test_table_dict", _test_table_dict);
+  }
+
+  virtual void TearDown() {
+    StorageManager::get().drop_table("test_table");
```

Comment:
you can use StorageManager::reset(); to ensure that the storage manager is reseted. (tests could add more tables)